spark/src/main/kotlin/com/adevinta/spark/components/textfields/SparkTextField.kt

```diff
@@ -442,16 +434,66 @@ private fun supportText(
     {
         // Prioritize the state message if there's one and fallback to the helper otherwise
         val message = state?.let { stateMessage } ?: helper
-        SupportingText(
-            text = message,
-            counterComposable = counterComposable,
-            statusIcon = stateIcon,
-        )
+        val stateMessageContentDescriptionModifier = if (state != null) {
+            val stateMessageContentDescription = computeStateMessageContentDescription(
+                state = state,
+                stateMessage = stateMessage,
+            )
+            Modifier.semantics { contentDescription = stateMessageContentDescription }
+        } else {
+            Modifier
+        }
+        Row(
+            modifier = Modifier.fillMaxWidth(),
+            horizontalArrangement = Arrangement.SpaceBetween,
+        ) {
+            stateIcon?.invoke(Modifier.padding(end = 4.dp))
+            Text(
+                modifier = Modifier
+                    .weight(1f, fill = true)
+                    .then(other = stateMessageContentDescriptionModifier),
+                text = message.orEmpty(),
+            )
+            counterComposable?.invoke(Modifier.padding(start = 8.dp))
+        }
     }
 } else {
     null
 }
 
+@Composable
+private fun computeLabelContentDescription(
+    required: Boolean,
+    label: String?,
```

Comment:
nit: If the label is null, shouldn't we produce no String instead of an empty string? (related to the question below, whenchecking for null input)